src/box/schema.cc

```diff
@@ -429,6 +451,27 @@ schema_init(void)
 	sc_space_new(BOX_FUNC_INDEX_ID, "_func_index", key_parts, 2,
 		     &on_replace_func_index);
 
+	/* _gc_consumers - gc_consumers for replicas. */
+	key_parts[0].fieldno = 0; /* UUID */
+	key_parts[0].type = FIELD_TYPE_STRING;
+	key_parts[1].fieldno = 1; /* vclock */
+	key_parts[1].type = FIELD_TYPE_MAP;
+	key_parts[2].fieldno = 2; /* opts */
+	key_parts[2].type = FIELD_TYPE_MAP;
+	struct space_opts opts = space_opts_default;
+	opts.group_id = 1;
+	/*
+	 * We don't set our system trigger on _gc_consumers because:
+	 * 1. We mustn't limit depth of on_replace triggers here since
+	 *    we do _gc_consumers:replace() from _cluster:delete().
+	 * 2. We shouldn't acquire ddl on replace in _gc_consumers because
+	 *    they are done in background by relay and would abort user
+	 *    transactions for no apparent reason.
+	 */
+	sc_space_new_with_opts(BOX_GC_CONSUMERS_ID, "_gc_consumers", key_parts,
```

Comment:
`_vinyl_deferred_delete` is very different - there we don't create `key_def`, primary index and do not use any triggers, it's just like `/dev/null`.
`_gc_consumers` is like any other system space, but it is local and we don't need `on_replace_dd_system_space` there, so it would be about 30 lines of copy paste. I would leave it as it is now.

Reply:
Outdated: there is no surrogate space anymore.